games/speedrun/index.js

```diff
@@ -0,0 +1,459 @@
+          else this.handleMove(key)
+        }
+      }
+    }
+
+    p5.keyReleased = () => {
+      const key = p5.key
+      if (['ArrowUp', 'ArrowDown', 'ArrowLeft', 'ArrowRight'].includes(key)) {
+        this.keysPressed.delete(key)
+        if (this.state.autoMoving) {
+          this.state.autoMoving = false
+          this.state.autoMovePath = []
+          this.state.glitch.intensity = 0
+          if (!this.state.resurrection.active) {
+            this.state.helpMessage.show = false
+            this.state.helpMessage.startTime = p5.millis()
+          }
+        }
+      }
+    }
+  }
+
+  allArrowsPressed() {
+    return this.keysPressed.has('ArrowUp') && 
+           this.keysPressed.has('ArrowDown') && 
+           this.keysPressed.has('ArrowLeft') && 
+           this.keysPressed.has('ArrowRight')
+  }
+
+  init(canvas) {
+    const margin = this.state.margin
+    const cellSize = this.state.cellSize
+    const rows = Math.floor((canvas.height - 2 * margin) / cellSize)
+    const cols = Math.floor((canvas.width - 2 * margin) / cellSize)
+
+    this.state = { 
+      ...this.state, 
+      maze: this.generateMaze(rows, cols),
+      rows, cols,
+      playerPosition: { x: 0, y: 0 },
+      endPosition: { x: cols - 1, y: rows - 1 },
+      visitedCells: new Set(['0,0']),
+      autoMoving: false,
+      autoMovePath: [],
+      gameOver: false,
+      won: false,
+      helpMessage: { show: false, startTime: this.libs.p5.millis() },
+    }
+  }
+
+  generateMaze(rows, cols) {
+    let maze = Array(rows).fill().map(() => 
+      Array(cols).fill().map(() => ({ top: true, right: true, bottom: true, left: true, visited: false }))
+    )
+
+    const dfs = (r, c) => {
+      maze[r][c].visited = true
+      const directions = [
+        { dr: -1, dc: 0, wall: 'top', oppositeWall: 'bottom' },
+        { dr: 0, dc: 1, wall: 'right', oppositeWall: 'left' },
+        { dr: 1, dc: 0, wall: 'bottom', oppositeWall: 'top' },
+        { dr: 0, dc: -1, wall: 'left', oppositeWall: 'right' }
+      ]
+
+      for (let i = directions.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [directions[i], directions[j]] = [directions[j], directions[i]]
+      }
+
+      for (const dir of directions) {
+        const newR = r + dir.dr
+        const newC = c + dir.dc
+        if (newR >= 0 && newR < rows && newC >= 0 && newC < cols && !maze[newR][newC].visited) {
+          maze[r][c][dir.wall] = false
+          maze[newR][newC][dir.oppositeWall] = false
+          dfs(newR, newC)
+        }
+      }
+    }
+
+    dfs(0, 0)
+
+    return maze
+  }
+
+  handleMove(key) {
+    if (this.state.gameOver || this.state.resurrection.active) return
+
+    const pos = this.state.playerPosition
+    const maze = this.state.maze
+    
+    let moved = false
+    let newPos = {...pos}
+
+    if (key === 'ArrowUp' && pos.y > 0 && !maze[pos.y][pos.x].top) {
```

Comment:
by the way - in miRCo engine you can access currently pressed keys with API `this.inputs.isPressedLeft()` etc